src/user/models/user/user.service.ts

```diff
@@ -64,6 +65,29 @@ export class UserService {
     return this.userRepo.findOne({ where: { address }, relations: needsRelation ? ['userData', 'wallet'] : [] });
   }
 
+  async getLinkedUser(id: number, address: string, blockchain: string): Promise<User> {
+    return this.userRepo
+      .createQueryBuilder('user')
+      .select('linkedUser.*')
+      .leftJoin('user.userData', 'userData')
+      .leftJoin('userData.users', 'linkedUser')
+      .where('user.id = :id', { id })
+      .andWhere('linkedUser.address = :address AND linkedUser.blockchain = :blockchain', { address, blockchain })
+      .getRawOne<User>();
+  }
+
+  async getAllLinkedUsers(id: number): Promise<LinkedUserDto[]> {
+    return this.userRepo
+      .createQueryBuilder('user')
+      .select(
+        'linkedUser.address, linkedUser.blockchain, isSwitchable = CASE WHEN linkedUser.stakingBalance = 0 then 1 else 0 end',
+      )
+      .leftJoin('user.userData', 'userData')
+      .leftJoin('userData.users', 'linkedUser')
+      .where('user.id = :id', { id })
+      .getRawMany<LinkedUserDto>();
```

Comment:
We could replace this query by a simpler one and do the mapping to LinkedUser in Code. WDYT?

Reply:
I am a little bit torn on this one. I would keep it like that and improve it in a later stage if needed?